mdal/frmts/mdal_ply.cpp

```diff
@@ -0,0 +1,445 @@
+  }
+  chunks = split( line, ' ' );
+
+  //
+  // configure the vectors to hold the data
+  //
+  for ( size_t elid = 0; elid < elements.size(); ++elid )
+  {
+    MDAL::DriverPly::element element = elements[elid];
+    if ( element.name == "vertex" )
+    {
+      vertexCount = element.size;
+      vertices.resize( vertexCount );
+      for ( size_t i = 0; i < element.properties.size(); ++i )
+      {
+        if ( element.properties[i] != "x" &&
+             element.properties[i] != "y" &&
+             element.properties[i] != "z" &&
+             !element.list[i] )
+        {
+          vertexDatasets.push_back( new std::vector<double> );
+          vProp2Ds.push_back( element.properties[i] );
+        }
+      }
+    }
+    else if ( element.name == "face" )
+    {
+      faceCount = element.size;
+      faces.resize( faceCount );
+      for ( size_t i = 0; i < element.properties.size(); ++i )
+      {
+        if ( element.properties[i] != "vertex_indices" &&
+             !element.list[i] )
+        {
+          faceDatasets.push_back( new std::vector<double> );
+          fProp2Ds.push_back( element.properties[i] );
+        }
+      }
+    }
+    else if ( element.name == "edge" )
+    {
+      edgeCount = element.size;
+      edges.resize( edgeCount );
+      for ( size_t i = 0; i < element.properties.size(); ++i )
+      {
+        if ( element.properties[i] != "vertex1" &&
+             element.properties[i] != "vertex2" &&
+             !element.list[i] )
+        {
+          edgeDatasets.push_back( new std::vector<double> );
+          eProp2Ds.push_back( element.properties[i] );
+        }
+      }
+    }
+
+    //
+    // load the data
+    //
+    for ( size_t i = 0; i < element.size; ++i )
+    {
+      //
+      // set the line size - we will only deal with one list at the begining of the line
+      //
+      size_t nChunks = element.properties.size();
+      if ( element.list[0] ) nChunks += MDAL::toSizeT( chunks[0] );
+      if ( chunks.size() != nChunks )
+      {
+        MDAL::Log::error( MDAL_Status::Err_IncompatibleMesh, name(), meshFile + " contains invalid line : " + line );
+        return nullptr;
+      }
+
+      //
+      // Load the vertexes
+      //
+      if ( element.name == "vertex" )
+      {
+        Vertex &vertex = vertices[i];
+        vertex.x = MDAL::toDouble( chunks[MDAL::DriverPly::getIndex( element.properties, "x" )] );
+        vertex.y = MDAL::toDouble( chunks[MDAL::DriverPly::getIndex( element.properties, "y" )] );
+        vertex.z = MDAL::toDouble( chunks[MDAL::DriverPly::getIndex( element.properties, "z" )] );
+        for ( size_t j = 0; j < vProp2Ds.size(); ++j )
+        {
+          std::vector<double> &dataset = *vertexDatasets[j];
+          double value = MDAL::toDouble( chunks[MDAL::DriverPly::getIndex( element.properties, vProp2Ds[j] )] );
+          dataset.push_back( value );
+          vertexDatasets[j] = &dataset;
+        }
+      }
+
+      //
+      //load the trinagles
```

Comment:
typo

Reply:
Oops. Fixed in my source. It will get upload in the next commit.